Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -19,49 +24,666 @@ FlexFormattingContext::~FlexFormattingContext()
 {
 }
 
-void FlexFormattingContext::run(Box& box, LayoutMode layout_mode)
+struct FlexItem {
+    Box& box;
+    float flex_base_size { 0 };
+    float hypothetical_main_size { 0 };
+    float hypothetical_cross_size { 0 };
+    float target_main_size { 0 };
+    bool frozen { false };
+    Optional<float> flex_factor {};
+    float scaled_flex_shrink_factor { 0 };
+    float max_content_flex_fraction { 0 };
+    float main_size { 0 };
+    float cross_size { 0 };
+    float main_offset { 0 };
+    float cross_offset { 0 };
+    bool is_min_violation { false };
+    bool is_max_violation { false };
+};
+
+struct FlexLine {
+    Vector<FlexItem*> items {};
```

Comment:
`Vector<T>` has a default ctor, there's no need to explicitly initialise it like this (here and below).